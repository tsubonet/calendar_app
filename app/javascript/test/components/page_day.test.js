import React from 'react'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })
import PageDay from '../../components/link'

class App extends React.Component {
  constructor(props) {
    super(props)
  }
  render() {
    const { event } = this.props
    return (
      <div>
        <span onClick={event}>Click!</span>
      </div>
    )
  }
}
function sampleFunc() {
  alert('clickされました。')
}

describe('clickイベントのテスト', () => {
  test('Appコンポーネントにイベントを渡す', () => {
    const testMock = jest.fn()
    const subject = shallow(<App event={testMock} />)
    subject.find('span').simulate('click')
    expect(testMock).toHaveBeenCalled()
  })
})
