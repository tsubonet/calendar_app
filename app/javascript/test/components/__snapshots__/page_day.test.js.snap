// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exist record test click edit record 1`] = `
<PageDay
  actionPath="calendar#day"
  date={
    Object {
      "day": 24,
      "month": 1,
      "year": 2018,
    }
  }
  patchRecord={[MockFunction]}
  record={
    Object {
      "created_at": "2018-01-08T05:33:57.000Z",
      "done_on": "2018-01-24",
      "id": 56,
      "result": "limited",
      "updated_at": "2018-01-08T05:33:57.000Z",
    }
  }
>
  <div>
    <p>
      2018
      年
      1
      月
      24
      日(
      水
      )
    </p>
    <dl>
      <dt>
        <i
          className="fas fa-hand-point-down"
        />
         この日の結果は...
      </dt>
      <dd>
        <div
          className="result-image limited"
        >
          <div>
            <a
              className="delete-trigger"
              onClick={[Function]}
            >
              <i
                className="fas fa-trash-alt fa-2x"
              />
            </a>
          </div>
        </div>
      </dd>
    </dl>
    <div
      style={
        Object {
          "marginBottom": "10px",
        }
      }
    >
      <a
        className="edit-trigger"
        href="javascript:;"
        onClick={[Function]}
      >
        <i
          className="fas fa-angle-left"
        />
         結果を修正する
      </a>
    </div>
    <Link
      href="/month/2018/1"
    >
      <a
        href="/month/2018/1"
        onClick={[Function]}
      >
        <i
          className="fas fa-angle-left"
        />
         カレンダーにもどる
      </a>
    </Link>
  </div>
</PageDay>
`;

exports[`exist record test click edit record 2`] = `
<PageDay
  actionPath="calendar#day"
  date={
    Object {
      "day": 24,
      "month": 1,
      "year": 2018,
    }
  }
  patchRecord={[MockFunction]}
  record={
    Object {
      "created_at": "2018-01-08T05:33:57.000Z",
      "done_on": "2018-01-24",
      "id": 56,
      "result": "limited",
      "updated_at": "2018-01-08T05:33:57.000Z",
    }
  }
>
  <div>
    <p>
      2018
      年
      1
      月
      24
      日(
      水
      )
    </p>
    <dl>
      <dt>
        <i
          className="fas fa-hand-point-down"
        />
         この日の結果は...
      </dt>
      <dd>
        <div
          className="result-image limited"
        >
          <div>
            <a
              className="delete-trigger"
              onClick={[Function]}
            >
              <i
                className="fas fa-trash-alt fa-2x"
              />
            </a>
          </div>
        </div>
      </dd>
    </dl>
    <dl>
      <dt>
        <i
          className="fas fa-hand-point-down"
        />
         結果を選択する
      </dt>
      <dd>
        <ul>
          <li>
            <a
              className=""
              data-result="good"
              onClick={[Function]}
            >
              <span />
              <span>
                完璧！
              </span>
            </a>
          </li>
          <li>
            <a
              className="isSelected"
              data-result="limited"
              onClick={[Function]}
            >
              <span />
              <span>
                半分くらいできた
              </span>
            </a>
          </li>
          <li>
            <a
              className=""
              data-result="wakeup"
              onClick={[Function]}
            >
              <span />
              <span>
                起きただけ
              </span>
            </a>
          </li>
          <li>
            <a
              className=""
              data-result="bad"
              onClick={[Function]}
            >
              <span />
              <span>
                起きれなかった
              </span>
            </a>
          </li>
          <li>
            <a
              className=""
              data-result="sick"
              onClick={[Function]}
            >
              <span />
              <span>
                体調不良
              </span>
            </a>
          </li>
        </ul>
      </dd>
    </dl>
    <Link
      href="/month/2018/1"
    >
      <a
        href="/month/2018/1"
        onClick={[Function]}
      >
        <i
          className="fas fa-angle-left"
        />
         カレンダーにもどる
      </a>
    </Link>
  </div>
</PageDay>
`;
